File: code/csharp/ch07/luavm/api/LuaVm.cs
using luavm.binchunk;

namespace luavm.api
{
    public partial interface LuaState
    {
        int PC();
        void AddPC(int n);
        uint Fetch();
        void GetConst(int idx);
        void GetRK(int rk);
        LuaState New(int stackSize, Prototype proto);
    }
}

File: code/rust/ch08/src/api/lua_vm.rs
pub trait LuaVM: super::lua_state::LuaState {
    fn pc(&self) -> isize;
    fn add_pc(&mut self, n: isize);
    fn fetch(&mut self) -> u32;
    fn get_const(&mut self, idx: isize);
    fn get_rk(&mut self, rk: isize);
    fn register_count(&self) -> usize;
    fn load_vararg(&mut self, n: isize);
    fn load_proto(&mut self, idx: usize);
}
